src/main/java/org/springframework/data/neo4j/core/Neo4jTemplate.java

```diff
@@ -244,15 +245,21 @@ private Object convertIdValues(@Nullable Neo4jPersistentProperty idProperty, Obj
 				.with(neo4jMappingContext.getRequiredBinderFunctionFor((Class<T>) entityToBeSaved.getClass()))
 				.fetchAs(Long.class).one();
 
-		if (entityMetaData.hasVersionProperty() && !optionalInternalId.isPresent()) {
-			throw new OptimisticLockingFailureException(OPTIMISTIC_LOCKING_ERROR_MESSAGE);
+
+		if (!optionalInternalId.isPresent()) {
+			if (entityMetaData.hasVersionProperty()) {
+				throw new OptimisticLockingFailureException(OPTIMISTIC_LOCKING_ERROR_MESSAGE);
+			}
+			// defensive exception throwing
```

Comment:
?

Reply:
The optional value of the internal id should always exist besides that the optimistic locking fails.